python/cuspatial/cuspatial/core/binpreds/binpred_interface.py

```diff
@@ -0,0 +1,559 @@
+            Preprocess the left-hand and right-hand GeoSeries. This method is
+            implemented by the subclass.
+        _compute_predicate(self, lhs, rhs)
+            Compute the binary predicate between two GeoSeries. This method is
+            implemented by the subclass.
+        _postprocess(self, lhs, rhs, point_indices, op_result)
+            Postprocess the output GeoSeries to ensure that they are of the
+            correct type for the predicate. This method is implemented by the
+            subclass.
+
+        Examples
+        --------
+        >>> from cuspatial.core.binpreds.binpred_dispatch import (
+        ...     CONTAINS_DISPATCH
+        ... )
+        >>> from cuspatial.core.geoseries import GeoSeries
+        >>> from shapely.geometry import Point, Polygon
+        >>> predicate = CONTAINS_DISPATCH[(
+        ...     lhs.column_type, rhs.column_type
+        ... )](align=True, allpairs=False)
+        >>> lhs = GeoSeries([Polygon([(0, 0), (1, 1), (1, 0)])])
+        >>> rhs = GeoSeries([Point(0, 0), Point(1, 1)])
+        >>> print(predicate(lhs, rhs))
+        0    False
+        dtype: bool
+        """
+        super().__init__(**kwargs)
+        self.kwargs = kwargs
+
+    @abstractmethod
+    def __call__(self, lhs: "GeoSeries", rhs: "GeoSeries") -> Series:
+        """System call for the binary predicate. Calls the _call method,
+        which is implemented by the subclass. Executing the binary predicate
+        returns the results of the binary predicate as a GeoSeries.
+
+        Parameters
+        ----------
+        lhs : GeoSeries
+            The left-hand GeoSeries.
+        rhs : GeoSeries
+            The right-hand GeoSeries.
+
+        Returns
+        -------
+        GeoSeries
+            The results of the binary predicate.
+
+        Examples
+        --------
+        >>> from cuspatial.core.binpreds.binpred_dispatch import (
+        ...     CONTAINS_DISPATCH
+        ... )
+        >>> from cuspatial.core.geoseries import GeoSeries
+        >>> from shapely.geometry import Point, Polygon
+        >>> predicate = CONTAINS_DISPATCH[(
+        ...     lhs.column_type, rhs.column_type
+        ... )](align=True, allpairs=False)
+        >>> lhs = GeoSeries([Polygon([(0, 0), (1, 1), (1, 0)])])
+        >>> rhs = GeoSeries([Point(0, 0), Point(1, 1)])
+        >>> print(predicate(lhs, rhs))
+        0    False
+        dtype: bool
+        """
+        return self._call(lhs, rhs)
+
+    @abstractmethod
+    def _call(self, lhs: "GeoSeries", rhs: "GeoSeries") -> Series:
+        """Call the binary predicate. This method is implemented by the
+        subclass.
+
+        Parameters
+        ----------
+        lhs : GeoSeries
+            The left-hand GeoSeries.
+        rhs : GeoSeries
+            The right-hand GeoSeries.
+
+        Returns
+        -------
+        result : GeoSeries
+            A GeoSeries of boolean values indicating whether each feature in
```

Comment:
Boolean value only needs `cudf.Series` to store, not Geoseries.

Reply:
Thanks.